pkg/odo/cli/component/create.go

```diff
@@ -342,16 +340,34 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 			// Interactive mode
 			// Get component type, name and namespace from user's choice via interactive mode
 
-			// devfile.yaml is not present, user has to specify the component type
-			// Component type: We provide supported devfile component list then let you choose
-			if !util.CheckPathExists(DevfilePath) {
+			if len(co.devfileMetadata.devfilePath) != 0 {
+				file, err := os.Stat(co.devfileMetadata.devfilePath)
+				if err != nil {
+					return err
+				}
+
+				if file.IsDir() {
+					return errors.Errorf("%s exists but it's not file", co.devfileMetadata.devfilePath)
+				}
+			}
+
+			if !util.CheckPathExists(DevfilePath) && len(co.devfileMetadata.devfilePath) == 0 {
```

Comment:
```suggestion
			if !util.CheckPathExists(DevfilePath) && co.devfileMetadata.devfilePath == "" {
```

Reply:
Yep, I will use `x == ""` style to handle the empty comparison instead if `len(x) == 0`